### ISSUE TITLE
Feature request: override LimitNOFILE value when using Systemd and es_max_open_files is defined

### ISSUE THREAD

**Issue by ballesterosam** [elastic/ansible-elasticsearch#683]
Hi!

I have been facing a problem during an instalation with Ansible, because I need to change the value for LimitNOFILE. If Systemd is enabled, there is no way to override LimitNOFILE default value (65535).

As we can read in the **elasticsearch.j2** template:

```
# Specifies the maximum file descriptor number that can be opened by this process
# When using Systemd, this setting is ignored and the LimitNOFILE defined in
# /usr/lib/systemd/system/elasticsearch.service takes precedence
#MAX_OPEN_FILES=65535
{% if es_max_open_files is defined %}
MAX_OPEN_FILES={{es_max_open_files}}
{% endif %}
```

So, value on **es_max_open_files** is ignored with Systemd. My suggestion is, if **es_max_open_files** is defined, LimitNOFILE setting should be added in the file **override.conf** file in the same way that LimitMEMLOCK is included when **bootstrap.memory_lock** is defined. 

This is from **elasticsearch-config.yml**

```
- when: use_system_d and m_lock_enabled
  become: yes
  block:
    - name: Make sure destination dir exists
      file:
        path: "{{ sysd_config_file | dirname }}"
        state: directory
        mode: "755"

    - name: Copy specific ElasticSearch Systemd config file
      ini_file:
        path: "{{ sysd_config_file }}"
        section: Service
        option: LimitMEMLOCK
        value: infinity
        mode: "644"
      notify:
       - reload systemd configuration
       - restart elasticsearch
```

Thank you.

**Comment by jmlrt** [comment]
Hi @ballesterosam,
Thanks for reporting this issue.
Do you think you could provide a PR to fix it?

**Comment by ballesterosam** [comment]
Hi @jmlrt 
Yes, of course. I have tried to provide a PR, but I have had some problems with my GitHub user. I will fix them and I will provide it again ;).

**Comment by jmlrt** [comment]
Fixed by #697
